services/core/shared/pouchdb.js

```diff
@@ -31,44 +31,46 @@ const createDb = async (name, idxList = []) => {
 	const db = new PouchDB(name, { auto_compaction: true });
 
 	idxList.forEach(async idxName => {
-		logger.debug(`Setting up index ${idxName}...`);
+		if (typeof idxName === 'string') idxName = [idxName];
+		logger.debug(`Setting up index ${name}/${idxName.join('-')}...`);
+
 		await db.createIndex({
 			index: {
-				fields: [idxName],
+				fields: idxName,
 			},
 		});
 	});
 
 	return db;
 };
 
-const getDbInstance = async (collectionName) => {
+const getDbInstance = async (collectionName, idxList = []) => {
 	if (!connectionPool[collectionName]) {
 		const dbDataDir = `${config.db.directory}/${collectionName}`;
 		if (!fs.existsSync(dbDataDir)) fs.mkdirSync(dbDataDir, { recursive: true });
 
 		connectionPool[collectionName] = await createDb(
 			dbDataDir,
-			config.db.collections[collectionName].indexes,
+			[...config.db.collections[collectionName].indexes, ...idxList],
 		);
 		logger.info(`Opened PouchDB database: ${collectionName}`);
 	}
 
 	const db = connectionPool[collectionName];
 
 	const write = async (doc) => {
-		if (!doc._id) doc._id = doc.id;
+		if (!doc._id) doc._id = doc.id || doc.address;
 		return db.upsert(doc);
 	};
 
 	const writeOnce = async (doc) => {
-		if (!doc._id) doc._id = doc.id;
+		if (!doc._id) doc._id = doc.id || doc.address;
 		return db.putIfNotExists(doc);
 	};
 
 	const writeBatch = async (docs) => {
 		docs.map(doc => {
-			if (!doc._id) doc._id = doc.id;
+			if (!doc._id) doc._id = doc.id || doc.address;
```

Comment:
I think `doc.address` is too specific here. Please define `doc._id` in the account code instead.